t/serializer_mutable.t

```diff
@@ -1,90 +1,162 @@
 use strict;
 use warnings;
 
-use Test::More tests => 41;
 use Dancer2::Serializer::Mutable;
-use Plack::Test;
 use HTTP::Request::Common;
-use Encode;
-use JSON::MaybeXS;
-use YAML;
-use Ref::Util qw<is_coderef>;
+use Plack::Test;
+use Test::More tests => 10;
 
 {
+
     package MyApp;
     use Dancer2;
-    use Ref::Util qw<is_hashref>;
+
+    our $data = {foo => "bar"};
+    our $return = \$data;
 
     set serializer => 'Mutable';
 
-    get '/serialize'     => sub { +{ bar => 'baz' } };
-    post '/deserialize'  => sub {
-        return request->data &&
-               is_hashref( request->data ) &&
-               request->data->{bar} ? { bar => request->data->{bar} } : { ret => '?' };
+    post '/' => sub {
+        Test::More::is_deeply request->data => $data,
+          "request->data"
+          or Test::More::explain(Test::More::explain request->data);
+        return $$return;
     };
 }
 
-my $app = MyApp->to_app;
-ok( is_coderef($app), 'Got app' );
+isa_ok my $app = MyApp->to_app => 'CODE', "app";
+
+use Plack::Middleware::TrafficLog;
+$app = Plack::Middleware::TrafficLog->wrap(
+    $app,
+    logger => sub { return; warn "@_" },
+    with_body => 1,    # TODO find out why this fixes the test
```

Comment:
Has anyone an idea why this fixes the test? Without that line I get the following error and I fail to debug where it originates.

```
t/serializer_mutable.t .. 1/10 
    #   Failed test 'response status code'
    #   at t/serializer_mutable.t line 141.
    #          got: '500'
    #     expected: '200'

    #   Failed test 'response content type'
    #   at t/serializer_mutable.t line 143.
    #          got: 'text/plain'
    #     expected: 'text/x-yaml'

    #   Failed test 'response content'
    #   at t/serializer_mutable.t line 158.
    #          got: 'Internal Server Error
    # 
    # Bad Content-Length: maybe client disconnect? (6 bytes remaining) at /home/USER/perl5/perlbrew/perls/perl-5.22.1/lib/site_perl/5.22.1/Plack/Request.pm line 260.
    # '
    #     expected: '---
    # foo: bar
    # '
    # Looks like you failed 3 tests of 3.

#   Failed test 'unsupported 'Content-Type' and supported 'Accept' header'
#   at t/serializer_mutable.t line 160.
# Looks like you failed 1 test of 10.
t/serializer_mutable.t .. Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/10 subtests 

Test Summary Report
-------------------
t/serializer_mutable.t (Wstat: 256 Tests: 10 Failed: 1)
  Failed test:  8
  Non-zero exit status: 1
Files=1, Tests=10,  0 wallclock secs ( 0.02 usr  0.00 sys +  0.30 cusr  0.05 csys =  0.37 CPU)
Result: FAIL
```